File: 06a.php
<?php

$file = "06.dat";
$fin = fopen($file, "r");
$data = fread($fin, filesize($file));
$instruction_lines = preg_split('/\n/', $data, -1, PREG_SPLIT_NO_EMPTY);
foreach ($instruction_lines as $line) {
    $i = 0;
    foreach (str_split($line) as $character) {
        $frequency[$i][$character] = ($frequency[$i][$character] ?? 0) + 1;
        $i++;
    }
}

foreach ($frequency as $column) {
    $max = 0;
    $character_max = '';
    foreach ($column as $character => $value) {
        if ($value > $max) {
            $max = $value;
            $character_max = $character;
        }
    }
    echo $character_max;
}

echo "\n";

fclose($fin);

?>

File: 06a.py
#!/usr/bin/python3

with open('06.dat') as f:
    data = f.read()
instruction_lines = data.split('\n')
instruction_lines.pop()
frequency = [dict() for _ in range(len(instruction_lines[0]))]
for line in instruction_lines:
    i = 0
    for character in line:
        if (character in frequency[i].keys()):
            frequency[i][character] = (frequency[i][character]) + 1
        else:
            frequency[i][character] = 1
        i += 1

for column in frequency:
    max = 0
    character_max = None
    for character in column.keys():
        if column[character] > max:
            max = column[character]
            character_max = character
    print(character_max, end='')
print('')
